File: Printpress.Application/Inventory/DTO/InventoryItemDto.cs
using Printpress.Domain;
using System.Text.Json.Serialization;

namespace Printpress.Application;

public record InventoryItemDto
{
    public Guid Id { get; init; }
    public string Name { get; init; }

    [JsonConverter(typeof(JsonStringEnumConverter))]
    public InventoryItemCategoryEnum InventoryItemCategory { get; init; }

    public int? PacksPerCarton { get; init; }
    public int? UnitsPerPack { get; init; }
    public int ExpectedPurchaseLossPercent { get; init; }
    public int ExpectedProductionWastePercent { get; init; }
}

File: Printpress.UI/Printpress_Angular/src/app/features/inventory/models/inventory-item.dto.ts
export interface InventoryItemDto {
  id: string;
  name: string;
  inventoryItemCategory: string;
  packsPerCarton: number | null;
  unitsPerPack: number | null;
  expectedPurchaseLossPercent: number;
  expectedProductionWastePercent: number;
  stockQuantity: number;
  totalInQuantity: number;
  totalOutQuantity: number;
  hasTransactions: boolean;
  isActive: boolean;
}
